newIDE/app/src/EventsFunctionsExtensionEditor/EventsFunctionConfigurationEditor/EventsFunctionParametersEditor.js

```diff
@@ -72,6 +74,80 @@ const validateParameterName = (i18n: I18nType, newName: string) => {
   return true;
 };
 
+type StrngSelectorEditorProps = {|
+  extraInfo: string,
+  setExtraInfo: newExtraInfo => void,
+|};
+
+const StringSelectorEditor = ({
+  extraInfo,
+  setExtraInfo,
+}: StrngSelectorEditorProps) => {
+  const [array, setArray] = React.useState([]);
+  const forceUpdate = useForceUpdate();
+  React.useEffect(
+    () => {
+      let parsedArray = null;
+      try {
+        parsedArray = JSON.parse(extraInfo);
+      } catch (e) {
+        console.warn('Cannot parse parameter extraInfo: ', e);
+      }
+
+      if (parsedArray !== null) {
+        setArray(parsedArray);
+      }
+    },
+    [extraInfo, array]
+  );
+
+  const updateExtraInfo = () => {
+    setExtraInfo(JSON.stringify(array));
+    forceUpdate();
```

Comment:
You use a forceUpdate because you've been modifying the state without calling the setter (by mutating `array`). With my solution, `array` won't be a state anymore, just something that is computed in the render. So remove the forceUpdate... but in this case, how will this component knows that it must re-render? Thanks to its props :) 
I t will re-render whenever `extraInfo` is changed - which is the responsibility of the parent.

tl;dr: remove `forceUpdate`  here and add it to the parent callback instead.